### PR TITLE
Refactor/extract search logic from nfts part2

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-502](https://emurgo.atlassian.net/browse/YOMO-502)

[YOMO-502]: https://emurgo.atlassian.net/browse/YOMO-502?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Search/SearchContext.tsx [hunk 1]
```diff
@@ -18,6 +18,18 @@ export const useSearch = () => {
   return value
 }
 
+export const useSearchResult = ({target, searchBy, sort = true}) => {
```
Comment (banklesss) [inline src/Search/SearchContext.tsx]: What should be the type of `target` in this case?

I tried `Array<Record<string, unknown>>`, but I am getting ts errors below when using the object

Comment (banklesss) [reply src/Search/SearchContext.tsx]: @stackchain @MichalSzorad @wolverineks

Comment (wolverineks) [reply src/Search/SearchContext.tsx]: get rid of this hook. keep it simple and filter directly in the component

File: src/Nfts/Nfts.tsx [hunk 2]
```diff
@@ -23,10 +23,13 @@ export const Nfts = () => {
     },
   })
 
-  const {searchResult: nftsSearchResult, searchTerm: nftsSearchTerm} = useSearchResult({
-    target: nfts,
-    searchBy: 'name',
-  })
+  const {search: nftsSearchTerm} = useSearch()
+  const searchTermLowerCase = nftsSearchTerm.toLowerCase()
+  const filteredNfts =
```
Comment (wolverineks) [inline src/Nfts/Nfts.tsx]: extract this to a separate function, add a test

File: src/Nfts/filterNfts.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const filterNfts = (nftsSearchTerm, nfts) => {
```
Comment (wolverineks) [inline src/Nfts/filterNfts.ts]: always type the parameters

File: src/Nfts/filterNfts.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import {YoroiNft} from '../yoroi-wallets'
+import {filterNfts} from './filterNfts'
+
+describe('filterNfts', () => {
+  const nfts = [{name: 'CryptoWolf #1234'}, {name: 'Bored Monkey #4567'}, {name: 'Apple Blocks #7890'}] as YoroiNft[]
+
+  it('filters NFTs correctly with capitalized search term', () => {
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: case-insensitive

File: src/Nfts/filterNfts.test.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    const filteredNfts = filterNfts('APple bLOcks', nfts)
+    expect(filteredNfts).toHaveLength(1)
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: expect(filteredNfts).toEqual([appleBlocks])

File: src/Nfts/filterNfts.test.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    expect(filteredNfts[0].name).toEqual('Apple Blocks #7890')
+  })
+
+  it('returns empty array when nft array is empty', () => {
+    const filteredNfts = filterNfts('Bored Monkey #4567', [])
+    expect(filteredNfts).toHaveLength(0)
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: expect(filteredNfts).toEqual([])

File: src/Nfts/filterNfts.test.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import {YoroiNft} from '../yoroi-wallets'
+import {filterNfts} from './filterNfts'
+
+describe('filterNfts', () => {
+  const nfts = [{name: 'CryptoWolf #1234'}, {name: 'Bored Monkey #4567'}, {name: 'Apple Blocks #7890'}] as YoroiNft[]
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: const cryptoWolf = ...
const boredMonkey = ...
const appleBlocks = ...
const nfts: YoroiNft[] = [cryptoWolf, boredMonkey, appleBlocks]

File: src/Nfts/filterNfts.test.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  })
+
+  it('returns empty array when nft array is empty', () => {
+    const filteredNfts = filterNfts('Bored Monkey #4567', [])
+    expect(filteredNfts).toHaveLength(0)
+  })
+
+  it('returns all NFTs when search term is empty', () => {
+    const filteredNfts = filterNfts('', nfts)
+    expect(filteredNfts).toHaveLength(3)
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: expect(filteredNfts).toEqual(nfts)

File: src/Nfts/filterNfts.test.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  })
+
+  it('sorts NFTs by name', () => {
```
Comment (wolverineks) [inline src/Nfts/filterNfts.test.ts]: filtering shouldnt sort

File: src/Nfts/filterNfts.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import {YoroiNft} from '../yoroi-wallets'
+
+export const filterNfts = (nftsSearchTerm: string, nfts: YoroiNft[]): YoroiNft[] => {
+  const searchTermLowerCase = nftsSearchTerm.toLowerCase()
+  const filteredNfts =
+    searchTermLowerCase.length > 0 && nfts.length > 0
+      ? nfts.filter((nft) => nft.name.toLowerCase().includes(searchTermLowerCase))
+      : nfts
+  return filteredNfts.sort((NftA, NftB) => NftA.name.localeCompare(NftB.name))
```
Comment (wolverineks) [inline src/Nfts/filterNfts.ts]: dont sort in the filter

File: src/yoroi-wallets/mocks/wallet.ts [hunk 11]
```diff
@@ -292,20 +292,23 @@ const fetchPoolInfo = {
   },
 }
 
+export const generateManyNfts = () => {
```
Comment (wolverineks) [inline src/yoroi-wallets/mocks/wallet.ts]: type

**Pull request closed without merging**